dynamiq/memory/backend/qdrant.py

```diff
@@ -0,0 +1,81 @@
+from qdrant_client.http import models as qdrant_models
+
+from dynamiq.components.embedders.base import BaseEmbedder
+from dynamiq.connections import Qdrant as QdrantConnection
+from dynamiq.memory.backend.base import Backend
+from dynamiq.prompts import Message
+
+
+class Qdrant(Backend):
+    """Qdrant implementation of the memory storage backend."""
+
+    def __init__(self, connection: QdrantConnection, embedder: BaseEmbedder, collection_name: str = "conversations"):
+        """Initializes the Qdrant memory storage."""
+        self.connection = connection
+        self.collection_name = collection_name
+        self.embedder = embedder
+
+        try:
+            self.client = self.connection.connect()
+        except Exception as e:
+            raise ConnectionError(f"Failed to connect to Qdrant: {e}")
+
+        # Check if the collection exists, create it if not
+        try:
+            self.client.get_collection(collection_name=self.collection_name)
+        except Exception as e:
```

Comment:
is it possible to use isinstance() to check the type of execption?

Reply:
Yes, I have added a few checks and they are sufficient.